frame/contracts/src/storage/meter.rs

```diff
@@ -412,12 +415,12 @@ where
 	/// change. This is the case when a `delegate_dependency` is added or removed, or when the
 	/// `code_hash` is updated. [`Self::charge`] cannot be used here because we keep track of the
 	/// deposit charge separately from the storage charge.
-	pub fn charge_deposit(&mut self, deposit_account: DepositAccount<T>, amount: DepositOf<T>) {
+	pub fn charge_deposit(&mut self, contract: T::AccountId, amount: DepositOf<T>) {
 		self.total_deposit = self.total_deposit.saturating_add(&amount);
-		self.charges.push(Charge { deposit_account, amount, terminated: false });
+		self.charges.push(Charge { contract, amount, terminated: false });
 	}
 
-	/// Charges from `origin` a storage deposit for contract instantiation.
+	/// Charge from `origin` a storage deposit plus the ed for contract instantiation.
```

Comment:
We transfer ed then charge deposit minus ed, so this comment or the code seems wrong 🤔

Reply:
`charge` is in itself a `transfer`, so I would not distinguish from transfer and charge. In fact we are taking "storage deposit + ed" from the origin.
How can we rephrase this to make it more clear?

Reply:
> In fact we are taking "storage deposit + ed" from the origin.

are you?
you do 
```rust
T::Currency::transfer(origin, contract, ed, Preservation::Preserve)?;
//...
E::charge(origin, contract, &deposit.saturating_sub(&Deposit::Charge(ed)), false)?;
```
you take `ed` and then `deposit - ed`, am I overlooking something here?

also since we transfer ed directly, couldn't we then use `fn charge_deposit` to delay the charge until later?

Reply:
> you take `ed` and then `deposit - ed`, am I overlooking something here?

yes, you're right

> also since we transfer ed directly, couldn't we then use `fn charge_deposit` to delay the charge until later?

absolutely, good point